examplepb/v1/example.proto

```diff
@@ -56,6 +56,11 @@ message Store {
 }
 
 message User {
+  option (dynamo.v1.msg).shard = {
```

Comment:
A few things:

Should this be a property of `dynamo.v1.msg.key`?

I'm not sure I believe in `shard_id` as a field, I was sort of imagining that it would opaque. We'll want/need to generate some functions anyway to iterate over the shard ids for scanning and other helpers, we could expose/calculate it that way. 

We should investigate a linter or something that does not let us remove this config once it is set.